pkg/blockbuilder/scheduler/config.go

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package scheduler
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/grafana/mimir/pkg/storage/ingest"
+)
+
+type Config struct {
+	BuilderConsumerGroup   string        `yaml:"builder_consumer_group"`
+	SchedulerConsumerGroup string        `yaml:"scheduler_consumer_group"`
+	KafkaMonitorInterval   time.Duration `yaml:"kafka_monitor_interval"`
+
+	// Config parameters defined outside the block-builder-scheduler config and are injected dynamically.
+	Kafka ingest.KafkaConfig `yaml:"-"`
+}
+
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.BuilderConsumerGroup, "block-builder-scheduler.builder-consumer-group", "block-builder", "The Kafka consumer group used by block-builders.")
```

Comment:
Not sure what this will be used for. I didn't expect block builders to have a consumer group at all, given all offsets should be tracked by the scheduler. I suggest to remove this for now, and re-introduce it in a later PR if it will be effectively used (if so, why?).

Reply:
Ah yes. This is something @codesome and I discussed: in the v1 of the scheduler, we'll just have the block-builder workers continue to checkpoint offsets to Kafka. For 2 reasons:

1. get v1 out quicker
2. less risky for backlogs: if there is a large backlog that takes 6 hours to consume, we'll get smaller section-granularity checkpoints rather than everything riding on that 6 hour job to complete.

(But it's mostly about reason number 1.) Let me know your thoughts.